tokio/src/runtime/builder.rs

```diff
@@ -168,7 +172,36 @@ impl Builder {
     /// # }
     /// ```
     pub fn thread_name(&mut self, val: impl Into<String>) -> &mut Self {
-        self.thread_name = val.into();
+        let val = val.into();
+        self.thread_name = Arc::new(Box::new(move || val.clone()));
+        self
+    }
+
+    /// Set the name fn of threads spawned by the `Runtime`'s thread pool.
```

Comment:
nit: maybe something like
```suggestion
    /// Sets a function used to generate the name of threads spawned by the `Runtime`'s thread pool.
```
is a little clearer?